### DISCUSSION TITLE
STM32H723ZG MQTT stop working

### DISCUSSION THREAD

**Discussion by MatteoRampazzo** [cesanta/mongoose#2781] [Ask questions, get answers] [answered]
Hi.
I’m trying to solve an issues with my board using mongoose to publish some data trough Mqtt.
My code is pretty easy, when a CAN message is received, into the callback the same message is published via MQTT.
The configuration seems working, in fact the board connect correctly to the broker, and send the messages.
The problem starts when a big amount of data is sent (about 1000 msg/s, but sometimes also a lot less), the messages stop being published, the row of code is normally executed, the program still run, but for some time no messages are published.
I’ve tried also without can callback, but simply publishing data during the while loop, obtaining the same situation.
I’ve tried also to run the same code on my pc (obv. changing the tcp/ip driver in the code) and there aren't problem, with this setup I’m able to send up to 10k msg/s without blocks.
To analyze the problem with my stm32, I’ve runned Wireshark on my pc connected to the same internet network, and I’ve seen that, when the code stop publish an error “tcp dup ack” is detected, by I’ve not find any more useful informations.

Let me know if I can add more details or file to help. Thanks a lot!

**Comment by scaprile** [comment, marked as answer]
- Mongoose does not actually send TCP-based data (like MQTT) when you call a function, but when you call the event manager. Your calls will buffer data that will be sent when the manager runs.
- 1000 msgs/s is 1 msg every millisecond, you'll need a heck of a high-speed link to achieve that in a sustained way. If you are using a very old version of our buiit-in TCP/IP stack, those, due to being experimental, used to close connections on errors.

If there is an actual issue, please open an issue, _honor the issue template_ so we can know your scenario and reproduce the issue if necessary, and let us know _exactly_ what you are doing
